src/test/java/com/google/tagpost/TagpostServerTest.java

```diff
@@ -0,0 +1,47 @@
+package com.google.tagpost;
+
+import static org.junit.Assert.assertEquals;
+
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.testing.GrpcCleanupRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for TagpostServer */
+@RunWith(JUnit4.class)
+public class TagpostServerTest {
+  /** Automatic graceful shutdown registered servers and channels at the end of test. */
+  @Rule public final GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+  /**
+   * To test the server, make calls with a real stub using the in-process channel, and verify
+   * behaviors or state changes from the client side.
+   */
+  @Test
+  public void replyMessage() throws Exception {
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
+
+    // Create a server, add service, start, and register for automatic graceful shutdown.
+    grpcCleanup.register(
```

Comment:
In fact we do not need to do this because we let the actual working function out of the framework, so we only need to test that function. But this one is fine, too.